lib/src/value/calculation.dart

```diff
@@ -287,6 +768,20 @@ class SassCalculation extends Value {
         "${pluralize('was', args.length, plural: 'were')} passed.");
   }
 
+  /// Returns a [Callable] named [name] that calls a single argument
+  /// math function.
+  static Value _singleArgument(
+      String name, Object argument, SassNumber mathFunc(SassNumber value),
+      {bool requireUnits = false, bool requireKnownUnits = false}) {
+    argument = _simplify(argument);
+    if (argument is! SassNumber ||
+        (requireKnownUnits && argument.hasUnit('%'))) {
+      return SassCalculation._(name, [argument]);
+    }
+    if (!requireUnits) argument.assertNoUnits();
```

Comment:
`requireUnits` doesn't really express what this is doing (this may have been a bad suggestion on my part). It's more accurate to say something like `if (forbidUnits) argument.assertNoUnits()`. That said, why aren't we checking for no units in `mathFunc()` itself?

Reply:
We are asserting no units in the math func itself as well. In a previous version of this file, we only did `assertNoUnits()` in the math function. But in a previous comment you mentioned we should throw an error from this file as well if the parameters contained units.